oaknational/Oak-Web-Application
# LESQ-1091: show new only filter on search page

## Description

Music year: 1985

- Adds UI to search page to filter 
- Updates test file
- Updates query when selected to be `curriculum=new`

## Issue(s)

Fixes #LESQ-1091

## How to test

1. Go to https://deploy-preview-2857--oak-web-application.netlify.thenational.academy
2. Use search
3. When the curriculum new button is selected, `curriculum=new` added to the url and all search items should be non-legacy. 

## Screenshots

How it should now look:

https://github.com/user-attachments/assets/2bef5db5-52c0-4454-a746-a6fe2ed7ee2c



## Checklist
